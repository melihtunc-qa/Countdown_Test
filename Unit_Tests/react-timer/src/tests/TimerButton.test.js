import React from 'react';
import { shallow } from 'enzyme';
import TimerButton from '../components/TimerButton/TimerButton';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('TimerButton Component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<TimerButton buttonAction={() => {}} buttonValue="Start" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('invokes buttonAction when clicked', () => {
    const mockButtonAction = jest.fn();
    const wrapper = shallow(<TimerButton buttonAction={mockButtonAction} buttonValue="Start" />);

    wrapper.find('.button-container').simulate('click');
    expect(mockButtonAction).toHaveBeenCalled();
  });
});
