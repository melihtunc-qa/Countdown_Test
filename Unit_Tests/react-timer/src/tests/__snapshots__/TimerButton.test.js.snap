// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerButton Component renders without crashing 1`] = `ShallowWrapper {}`;
